src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Approve.tsx

```diff
@@ -118,7 +118,9 @@ export const Approve: React.FC<ApproveProps> = ({ accountId, onNext }) => {
         contractAddress,
         userAddress: accountAddress,
         wallet: walletState.wallet,
-        amount: bnOrZero(state?.deposit.cryptoAmount).times(`1e+${asset.precision}`).toString(),
+        amount: bn(
+          bnOrZero(state?.deposit.cryptoAmount).times(`1e+${asset.precision}`).integerValue(),
```

Comment:
wanna give the pow pow treatment while we're here?

Reply:
also is `integerValue` the secret sauce we've been missing to avoid this in a lot of places?

Reply:
> wanna give the pow pow treatment while we're here?

Might as well! d6d6755c5aa68dc67dc1fc33e40902643bb88266 keep in mind this is going away very very soon